fs/vfs/fs_eventfd.c

```diff
@@ -0,0 +1,623 @@
+
+  /* The following is a list if poll structures of threads waiting for
+   * driver events. The 'struct pollfd' reference for each open is also
+   * retained in the f_priv field of the 'struct file'.
+   */
+
+#ifdef CONFIG_EVENT_FD_POLL
+  FAR struct pollfd *d_fds[CONFIG_EVENT_FD_NPOLLWAITERS];
+#endif
+};
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int eventfd_open(FAR struct file *filep);
+static int eventfd_close(FAR struct file *filep);
+
+static ssize_t eventfd_do_read(FAR struct file *filep, FAR char *buffer,
+                               size_t len);
+static ssize_t eventfd_do_write(FAR struct file *filep,
+                                FAR const char *buffer, size_t len);
+static int eventfd_do_ioctl(FAR struct file *filep, int cmd,
+                            unsigned long arg);
+#ifdef CONFIG_EVENT_FD_POLL
+static int eventfd_poll(FAR struct file *filep, FAR struct pollfd *fds,
+                       bool setup);
+#endif
+
+static int eventfd_alloc_dev(eventfd_t counter, int flags);
+
+#ifdef CONFIG_EVENT_FD_POLL
+static void eventfd_pollnotify(FAR struct eventfd_priv_s *dev,
+                               pollevent_t eventset);
+#endif
+
+static int get_unique_minor(void);
+static void release_minor(int minor);
+
+static FAR struct eventfd_priv_s *eventfd_allocdev(void);
+static void eventfd_destroy(FAR struct eventfd_priv_s *dev);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const struct file_operations g_eventfd_fops =
+{
+  eventfd_open,     /* open */
+  eventfd_close,    /* close */
+  eventfd_do_read,  /* read */
+  eventfd_do_write, /* write */
+  0,                /* seek */
+  eventfd_do_ioctl  /* ioctl */
+#ifdef CONFIG_EVENT_FD_POLL
+  , eventfd_poll    /* poll */
+#endif
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+FAR struct eventfd_priv_s *eventfd_allocdev(void)
+{
+  FAR struct eventfd_priv_s *dev;
+
+  dev = (FAR struct eventfd_priv_s *)
+    kmm_malloc(sizeof(struct eventfd_priv_s));
+  if (dev)
+    {
+      /* Initialize the private structure */
+
+      memset(dev, 0, sizeof(struct eventfd_priv_s));
+      nxsem_init(&dev->exclsem, 0, 0);
+      nxsem_init(&dev->rdsem, 0, 0);
+      nxsem_init(&dev->wrsem, 0, 0);
+
+      /* The read/write wait semaphores are used for signaling
+       * and should not have priority inheritance enabled.
+       */
+
+      nxsem_set_protocol(&dev->rdsem, SEM_PRIO_NONE);
+      nxsem_set_protocol(&dev->wrsem, SEM_PRIO_NONE);
+    }
+
+  return dev;
+}
+
+void eventfd_destroy(FAR struct eventfd_priv_s *dev)
```

Comment:
add static

Reply:
all statics in section:
/****************************************************************************
 * Private Function Prototypes
 ****************************************************************************/
 Maybe not the best way to do this

Reply:
Yes, even we add static for each function in "Private Function Prototypes" section, but the convention still add static in the implemetnation too.